### PR TITLE
POC base tracking migration

### PR DESCRIPTION
**Pull request by lboogie04**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
POC of Migrating Initializers in Asset Pipeline to Pack Tags

Only includes migrating `BaseTracking` for testing purposes.

## Related Tickets & Documents

- Related Issue #18828 
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by Ridhwana — COMMENTED** [review]
Hey @lboogie04, its so nice to see this improvement 😊 I pulled down and played around with the code a bit. I have a few follow up questions: 

1. Are there any particular aspects that you would you like feedback on within this PR?
2. Including the `baseTracking` in the `main_stories_feed` enables the Google Analytics from the feed page when the user is signed out. By adding the pack file to that page, were you just testing that the pack file gets called or are you seeing this as the final place where the pack file should live? I suspect that including it only in this view does not enable tracking to work across the site.

File: app/javascript/packs/baseTracking.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
+function initializeBaseTracking() {
+    console.log("*************Initializing base tracker from packs************************")
+    trackGoogleAnalytics3();
+    trackGoogleAnalytics4();
+    trackCustomImpressions();
+  }
+  
+  function trackGoogleAnalytics3() {
+    let wait = 0;
```
Comment (fdocr) [inline app/javascript/packs/baseTracking.js:9]: Could you please fix this indentation detail? Starting on line 2 there's a "double tab" along the file all the way to the end.

**Review by fdocr — COMMENTED** [review]
> And as for baseTracking I simply want to assure it's getting called. I'll most likely leave baseTracking where it is and abstract some of the remaining logic.

I might be missing/misunderstanding something. Is the end goal here to completely remove `app/assets/javascripts/initializer/initializeBaseTracking.js.erb` and have that live in this new `app/javascript/packs/baseTracking.js`?

If I remove [this line from `initializePage.js`](https://github.com/forem/forem/blob/e63236363ffd4cc8f35a2935e72aa406d931cc73/app/assets/javascripts/initializePage.js#L18) and `initializeBaseTracking.js.erb` (asset pipeline file & initializer call) then we would have the new `baseTracking.js` code execute, but it isn't locally for me AFAICT.

Like Ridhwana mentioned, I think that including `"baseTracking"` to `javascript_packs_with_chunks_tag` needs to be added to [application.html.erb (likely here)](https://github.com/forem/forem/blob/e63236363ffd4cc8f35a2935e72aa406d931cc73/app/views/layouts/application.html.erb#L22-L26) layout so it's included/executed on all of the site.

Does this make sense?

In terms of standards I only left one inline comment that I noticed 🙂

**Comment by Ridhwana** [comment]
> So I'd definitely like feedback from you from a Standards perspective and assure it aligns with paradigms that you may be setting forth.

In terms of standards, this looks good @lboogie04, moving to the packs files aligns with the standards that we're trying to implement 🎉

> For further clarity, moving the pack tag to application.html.erb was the approach I'm taking I'm ONLY testing that the pack is initiating for the page/partial I've added it to.

Also thank you for clarifying the above.

**Pull request closed without merging**